mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -0,0 +1,416 @@
+    )
+    val parents: Parents = selector match
+      case EmptyTree =>
+        val seenFromType = parent match
+          case TreeApply(fun, _) if fun.tpe != null && !fun.tpe.isErroneous =>
+            fun.tpe
+          case _ =>
+            parent.tpe
+        seenFromType.paramInfoss match
+          case (head :: Nil) :: _
+              if definitions.isFunctionType(head) || head.isRef(
+                definitions.PartialFunctionClass
+              ) =>
+            val dealiased = head.widenDealias
+            val argTypes =
+              head.argTypes.init
+            new Parents(argTypes, definitions)
+          case _ =>
+            new Parents(NoType, definitions)
+      case sel => new Parents(sel.tpe, definitions)
+
+    val result = ListBuffer.empty[CompletionValue]
+    val isVisited = mutable.Set.empty[Symbol]
+
+    def visit(sym: Symbol, name: String, autoImports: List[l.TextEdit]): Unit =
+      val isValid = !isVisited(sym) && !parents.isParent(sym)
+        && (sym.is(Case) || sym.is(Flags.Module) || sym.isClass)
+        && parents.isSubClass(sym, false)
+      def recordVisit(s: Symbol): Unit =
+        if s != NoSymbol && !isVisited(s) then
+          isVisited += s
+          recordVisit(s.moduleClass)
+          recordVisit(s.sourceModule)
+      if isValid then
+        recordVisit(sym)
+        result += completionGenerator.toCompletionValue(
+          sym,
+          name,
+          autoImports,
+        )
+      end if
+    end visit
+    val selectorSym = parents.selector.typeSymbol
+
+    // Step 1: walk through scope members.
+    indexedContext.scopeSymbols
+      .foreach(s => visit(s.info.dealias.typeSymbol, s.decodedName, Nil))
+
+    // Step 2: walk through known subclasses of sealed types.
+    selectorSym.sealedStrictDescendants.foreach { sym =>
+      if !(sym.is(Sealed) && (sym.is(Abstract) || sym.is(Trait))) then
+        val autoImport = autoImportsGen.forSymbol(sym)
+        autoImport match
+          case Some(value) =>
+            visit(sym.info.dealias.typeSymbol, sym.decodedName, value)
+          case scala.None =>
+            visit(sym.info.dealias.typeSymbol, sym.showFullName, Nil)
+      else ()
+    }
+
+    // Step 3: special handle case when selector is a tuple or `FunctionN`.
+    if definitions.isTupleClass(selectorSym) || definitions.isFunctionClass(
+        selectorSym
+      )
+    then
+      val label =
+        if !patternOnly then s"case ${parents.selector.show} =>"
+        else parents.selector.show
+      result += CompletionValue.CaseKeyword(
+        selectorSym,
+        label,
+        Some(
+          if !patternOnly then
+            if config.isCompletionSnippetsEnabled() then "case ($0) =>"
+            else "case () =>"
+          else if config.isCompletionSnippetsEnabled() then "($0)"
+          else "()"
+        ),
+        Nil,
+        range = Some(completionPos.toEditRange),
+        command = config.parameterHintsCommand().asScala,
```

Comment:
What is this for?
It looks like this setting triggers parameter hint command https://scalameta.org/metals/docs/integrations/new-editor/#compileroptionsparameterhintscommand but, as far as I know, case/match completion moves the cursor to the position where parameter hint is not available (e.g. c`ase Some(value) => @@`)

Are there any examples where this command is useful?

Reply:
> What is this for?
It looks like this setting triggers parameter hint command https://scalameta.org/metals/docs/integrations/new-editor/#compileroptionsparameterhintscommand but, as far as I know, case/match completion moves the cursor to the position where parameter hint is not available (e.g. case Some(value) => @@)

In this case we move cursor to `case (@@) =>`

>Are there any examples where this command is useful?

Yes, situations like `List(1 -> 2).map{ cas@@ }` (tests `brace` and `brace-function2` in `CompletionCaseSuite`